src/app/clusters/tls-client-management-server/TlsClientManagementCluster.cpp

```diff
@@ -126,110 +145,135 @@ TlsClientManagementCluster::EncodeProvisionedEndpoints(EndpointId matterEndpoint
-                                              req.endpointID, [&](auto & endpoint) -> CHIP_ERROR {
-                                                  Commands::FindEndpointResponse::Type response;
-                                                  response.endpoint = endpoint;
-                                                  ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
-                                                  return CHIP_NO_ERROR;
-                                              });
+        mDelegate.FindProvisionedEndpointByID(endpointId, fabric, req.endpointID, [&](auto & endpoint) -> CHIP_ERROR {
+            Commands::FindEndpointResponse::Type response;
+            response.endpoint = endpoint;
+            ConcreteCommandPath responsePath(mPath.mEndpointId, TlsClientManagement::Id, Commands::FindEndpoint::Id);
+            commandHandler.AddResponse(responsePath, response);
+            return CHIP_NO_ERROR;
+        });
 
     if (result == CHIP_ERROR_NOT_FOUND)
     {
-        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+        return Status::NotFound;
     }
-    else if (result != CHIP_NO_ERROR)
+
+    if (result != CHIP_NO_ERROR)
     {
-        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+        return Status::Failure;
     }
+
+    return std::nullopt;
 }
 
-void TlsClientManagementCluster::HandleRemoveEndpoint(HandlerContext & ctx, const Commands::RemoveEndpoint::DecodableType & req)
+std::optional<DataModel::ActionReturnStatus>
+TlsClientManagementCluster::HandleRemoveEndpoint(CommandHandler & commandHandler,
+                                                 const Commands::RemoveEndpoint::DecodableType & req)
 {
     ChipLogDetail(Zcl, "TlsClientManagement: RemoveEndpoint");
 
-    VerifyOrReturn(req.endpointID <= kMaxTlsEndpointId, ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError));
+    if (req.endpointID > kMaxTlsEndpointId)
+    {
+        return Status::ConstraintError;
+    }
+
+    auto fabric     = commandHandler.GetAccessingFabricIndex();
+    auto endpointId = mPath.mEndpointId;
 
-    auto status = mDelegate.RemoveProvisionedEndpointByID(ctx.mRequestPath.mEndpointId,
-                                                          ctx.mCommandHandler.GetAccessingFabricIndex(), req.endpointID);
+    auto status = mDelegate.RemoveProvisionedEndpointByID(endpointId, fabric, req.endpointID);
 
     if (status == Status::Success)
     {
-        MatterReportingAttributeChangeCallback(ctx.mRequestPath.mEndpointId, TlsClientManagement::Id,
-                                               TlsClientManagement::Attributes::ProvisionedEndpoints::Id);
+        NotifyAttributeChanged(TlsClientManagement::Attributes::ProvisionedEndpoints::Id);
     }
 
-    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+    return DataModel::ActionReturnStatus(status);
```

Comment:
```suggestion
    return status;
```

We also have https://github.com/project-chip/connectedhomeip/blob/master/src/app/server-cluster/DefaultServerCluster.h#L121

So maybe instead of the if and retur, you could do

```cpp
return NotifyAttributeChangedIfSuccess(TlsClientManagement::Attributes::ProvisionedEndpoints::Id, status);
```